python/adelphi/adelphi/nb.py

```diff
@@ -47,31 +60,36 @@ def build_select_statements(keyspace, table):
 
 
 def build_insert_statements(keyspace, table):
-    cols = table.columns.values()
-    primary_keys = set(table.primary_key)
-    col_names = ",".join([quote_str(col.name) for col in cols])
-    def binding_name(col):
-        return seq_binding_name(col) if col in primary_keys else dist_binding_name(col)
-    col_bindings = ",".join(["{" + binding_name(c) + "}" for c in cols])
-    return "insert into {}.{} ({}) values ({})".format(quote_str(keyspace.name), quote_str(table.name), col_names, col_bindings)
+    (pk_cols, plain_cols) = partition_cols(table)
+    col_names = [c.name for c in (table.primary_key + plain_cols) if is_supported_type(c)]
+
+    pk_bindings = ["{" + seq_binding_name(c) + "}" for c in table.primary_key if is_supported_type(c)]
+    plain_bindings = ["{" + dist_binding_name(c) + "}" for c in plain_cols if is_supported_type(c)]
+    col_bindings = ",".join(pk_bindings + plain_bindings)
 
+    return "insert into {}.{} ({}) values ({})".format(quote_str(keyspace.name), quote_str(table.name), ",".join(col_names), col_bindings)
 
-class ColumnTypeException(Exception):
-    """Exception indicinating an error in the handling of a specific column type"""
+
+class UnsupportedPrimaryKeyTypeException(Exception):
+    """Exception indicating a primary key column for which there is no sequence support"""
     pass
 
 
 class NbExporter(BaseExporter):
 
     def __init__(self, cluster, props):
 
+        self.rampup_cycles = props["rampup-cycles"]
+        self.main_cycles = props["main-cycles"]
+        self.numeric_max = min((self.rampup_cycles + self.main_cycles) * 1000, MAX_NUMERIC_VAL)
```

Comment:
Seemed reasonable to impose a global absolute max on the size of a number within our flow here.  The logic here is the minimum of either our absolute max or a range of 1000 values for each specified cycle (for the distribution case, which requires a wider range of values than the sequence case)